src/nodes/display/PixelationNode.js

```diff
@@ -0,0 +1,244 @@
+import TempNode from '../core/TempNode.js';
+import { uv } from '../accessors/UVNode.js';
+import { addNodeElement, tslFn, nodeObject, vec2, vec3, float, If } from '../shadernode/ShaderNode.js';
+import { NodeUpdateType } from '../core/constants.js';
+import { uniform } from '../core/UniformNode.js';
+import { dot, clamp, smoothstep, sign, step, floor } from '../math/MathNode.js';
+import { Vector4 } from '../../math/Vector4.js';
+import { property } from '../core/PropertyNode.js';
+import QuadMesh from '../../renderers/common/QuadMesh.js';
+import { RenderTarget } from '../../core/RenderTarget.js';
+import { passTexture } from './PassNode.js';
+import { NearestFilter } from '../../constants.js';
+
+const createEdgesQuad = new QuadMesh();
+const lowerResolutionQuad = new QuadMesh();
+
+class PixelationNode extends TempNode {
+
+	constructor( textureNode, depthNode, normalNode, pixelSizeNode, normalEdgeStrength, depthEdgeStrength ) {
+
+		super();
+
+		// Input textures
+
+		this.textureNode = textureNode;
+		this.depthNode = depthNode;
+		this.normalNode = normalNode;
+
+		// Input uniforms
+
+		this.pixelSizeNode = pixelSizeNode;
+		this.normalEdgeStrength = normalEdgeStrength;
+		this.depthEdgeStrength = depthEdgeStrength;
+
+		// Private uniforms
+
+		this._resolution = uniform( new Vector4() );
+
+		// Intermediary render targets
+
+		this._createEdgesRT = new RenderTarget();
+		this._createEdgesRT.texture.name = 'PixelationNode.renderEdges';
+		this._createEdgesRT.texture.minFilter = NearestFilter;
+		this._createEdgesRT.texture.magFilter = NearestFilter;
+		this._lowerResolutionRT = new RenderTarget();
+		this._lowerResolutionRT.texture.name = 'PixelationNode.lowerResolution';
+		this._lowerResolutionRT.texture.minFilter = NearestFilter;
+		this._lowerResolutionRT.texture.magFilter = NearestFilter;
+
+		// Output textures
+
+		this._outputTextureNode = passTexture( this, this._lowerResolutionRT.texture );
+
+		this.updateBeforeType = NodeUpdateType.RENDER;
+
+	}
+
+	setSize( width, height ) {
+
+		this._createEdgesRT.setSize( width, height );
+		this._lowerResolutionRT.setSize( width, height );
+
+	}
+
+	updateBefore( frame ) {
+
+		const { renderer } = frame;
+
+		const textureNode = this.textureNode;
+		const map = textureNode.value;
+
+		// Set resolution uniform
+
+		const adjustedWidth = map.image.width / this.pixelSizeNode.value;
+		const adjustedHeight = map.image.height / this.pixelSizeNode.value;
```

Comment:
You must compute integers here (like in the original). This will also make the versions more consistent:
```js
const adjustedWidth = Math.floor( map.image.width / this.pixelSizeNode.value );
const adjustedHeight = Math.floor( map.image.height / this.pixelSizeNode.value );
```

Reply:
BTW: The produces its beauty, depth and normal buffers with these adjusted resolution values. I wonder if this bit is responsible for the final visual difference.

What I am referring to is some kind of offset of the entire scene that gets larger and this more noticeable by increasing the pixel size.

This offset is still something that can be considered as a bug a needs to be resolved.

Reply:
As per https://github.com/mrdoob/three.js/pull/28802#issuecomment-2212559639, my initial first draft port was what you described, directly resizing the render targets based on the adjusted width and height, but WebGPURenderer and WebGLRenderer seem to handle the final output of a render target differently.

Reply:
Would you mind implementing the solution again? I would like to investigate the visual differences.

In any even, we should produce the beauty, depth and normal buffers like in `RenderPixelatedPass` meaning with a lower resolution. Depending on the `pixelSize` value, we are able to produces these buffers with a much lower resolution than now which saves a lot of GPU bandwidth. Which have to keep this optimization when porting the effect over to `WebGPURenderer` and `PostProcessing`.

`RenderPixelatedPass` is an alternative to `RenderPass` so it might make sense to derive rename the node to `PixelationPassNode` and derive it from `PassNode`. Instead of using `pass()` in the example, it would be:
```js
const scenePass = pixelationPass( scene, camera );
```
The MRT configuration happens inside `PixelationPassNode` then.

Reply:
I'll convert the PR back to "draft". There are still conceptual issues to implement so a final review is not yet possible.